### PR TITLE
implement default value coercion and validation

### PR DESCRIPTION
**Pull request by andimarek**
This PR adds one new coercing function which aim to resolves a long outstanding regarding custom Scalars and default value coercion:

Currently there is no well defined way for custom Scalar values to be printed as SDL.
Related to that default values are not properly coerced.

This also applies to applied schema directives which also contains values.

# Breaking changes

## Argument/InputObject defaultValue/values
`GraphQLArgument.defaultValue/value` and `GraphQLInputObject.defaultValue` are renamed to `argumentDefaultValue/argumentValue` and `inputFieldDefaultValue`.

We did this to signal clearly that the returned values changed. See JavaDoc for details.

## Non standard scalars removed

We removed previously deprecated non standard Scalars (Long, short etc). These Scalars are still available via https://github.com/graphql-java/graphql-java-extended-scalars

## Removed deprecated Constructors

Removed previously deprecated constructors for `GraphQLArgument` and `GraphQLInputObjectField`

### CODE REVIEW

File: src/main/java/graphql/parser/Parser.java [hunk 1]
```diff
@@ -39,7 +47,29 @@ public Document parseDocument(String input, String sourceName) throws InvalidSyn
...
+            return new Object[]{documentContext, value};
+        };
+        MultiSourceReader multiSourceReader = MultiSourceReader.newMultiSourceReader()
+                .string(input, null)
+                .trackData(true)
+                .build();
+        return (Value<?>) parseImpl(multiSourceReader, nodeFunction);
+    }
+
+    private Node parseImpl(Reader reader, BiFunction<GraphqlParser, GraphqlAntlrToLanguage, Object[]> nodeFunction) throws InvalidSyntaxException {
```
Comment (bbakerman) [inline src/main/java/graphql/parser/Parser.java:72]: Nice trick

File: src/main/java/graphql/schema/validation/AppliedDirectiveArgumentsAreValid.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+import graphql.schema.GraphQLSchemaElement;
+import graphql.schema.GraphQLTypeVisitorStub;
+import graphql.schema.InputValueWithState;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+import graphql.validation.ValidationUtil;
+
+import static java.lang.String.format;
+
+public class AppliedDirectiveArgumentsAreValid extends GraphQLTypeVisitorStub {
```
Comment (bbakerman) [inline src/main/java/graphql/schema/validation/AppliedDirectiveArgumentsAreValid.java:20]: `@Internal `

Comment (bbakerman) [reply src/main/java/graphql/schema/validation/AppliedDirectiveArgumentsAreValid.java:20]: ?

**Review by bbakerman — APPROVED** [review]

File: src/main/java/graphql/schema/validation/AppliedDirectiveArgumentsAreValid.java (hunk 2 above)
Comment (andimarek) [reply src/main/java/graphql/schema/validation/AppliedDirectiveArgumentsAreValid.java:20]: yes.

**Comment by yaacovCR** [comment]
Hello @andimarek et al!

At the [latest WG meeting](https://github.com/graphql/graphql-wg/blob/main/agendas/2023/01-Jan/05-wg-primary.md), we again discussed advancing default value coercion/validation within the reference implementation.

Just touching base at @IvanGoncharov 's suggestion to see if there have been any lessons learned from the implementation here?

Anything working well -- or not so well -- in the 1.5 years since this has been live in `graphql-java`?

Any feedback/commentary that might inform how the spec/reference implementation should look?

**Pull request merged**